Lib/_pyrepl/unix_console.py

```diff
@@ -388,7 +389,15 @@ def get_event(self, block: bool = True) -> Event | None:
         while self.event_queue.empty():
             while True:
                 try:
-                    self.push_char(self.__read(1))
+                    char = self.__read(1)
+                except BlockingIOError:
+
+                    if not os.get_blocking(sys.stdin.fileno()):
+                        return None
+                    else:
+                        raise
+                try:
+                    self.push_char(char)
```

Comment:
I don't think this is correct. The OSError EINTR handled below is also raised by the `os.read` in `__read()`.

Reply:
Whoops, good catch. I will unify the two.